### PR TITLE
fix points write panic

### PR DESCRIPTION
**Pull request by oiooj**
fix `index out of range` risk, log here:

``` shell
[wal] 2015/10/28 11:39:55 write to index of partition 1 took 3.516488ms
[wal] 2015/10/28 11:40:05 Flush due to idle. Flushing 8 series with 8 points and 424 bytes from partition 1
[wal] 2015/10/28 11:40:05 write to index of partition 1 took 3.126415ms
[wal] 2015/10/28 11:40:15 Flush due to idle. Flushing 8 series with 8 points and 424 bytes from partition 1
[wal] 2015/10/28 11:40:15 write to index of partition 1 took 1.716188ms
[http] 2015/10/28 11:40:25 221.221.221.98 - - [28/Oct/2015:11:40:25 +0800] POST /write HTTP/1.1 400 57 - Mozilla/5.0 (Macintosh; Intel Mac 
OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.71 Safari/537.36 9facaff5-7d25-11e5-8001-000000000000 2.522487ms
[wal] 2015/10/28 11:40:25 Flush due to idle. Flushing 8 series with 8 points and 424 bytes from partition 1
[wal] 2015/10/28 11:40:25 write to index of partition 1 took 12.511171ms
[wal] 2015/10/28 11:40:35 Flush due to idle. Flushing 8 series with 8 points and 442 bytes from partition 1
[wal] 2015/10/28 11:40:35 write to index of partition 1 took 7.997816ms
[http] 2015/10/28 11:40:38 221.221.221.98 - - [28/Oct/2015:11:40:38 +0800] POST /write HTTP/1.1 400 45 - Mozilla/5.0 (Macintosh; Intel Mac 
OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.71 Safari/537.36 a776548d-7d25-11e5-8002-000000000000 19.391701ms
[wal] 2015/10/28 11:40:45 Flush due to idle. Flushing 8 series with 8 points and 442 bytes from partition 1
[wal] 2015/10/28 11:40:45 write to index of partition 1 took 1.890025ms
[http] 2015/10/28 11:40:50 221.221.221.98 - - [28/Oct/2015:11:40:50 +0800] POST /write HTTP/1.1 200 23 - Mozilla/5.0 (Macintosh; Intel Mac 
OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.71 Safari/537.36 af16a5b9-7d25-11e5-8003-000000000000 1.222898ms [pan
ic:runtime error: index out of range]

```

### CODE REVIEW

**Comment by otoolep** [comment]
@oiooj -- can you please run the test suite locally before pushing up any further commits? That way you will get the quickest turnaround when debugging your code.

**Comment by oiooj** [comment]
@otoolep  sure, I have ran the test file. 

``` shell
$ go test points_test.go 
ok      command-line-arguments  0.013s

$ go test handler_test.go 
ok      command-line-arguments  0.019s

```

It is right ?

**Comment by otoolep** [comment]
I don't know, depends if you are are in the correct directory.
```
$ ls
balancer.go       circle-test.sh  cmd                       DOCKER.md  influxql                    Makefile  nightly.sh  README.md  snapshot  test-32bit-docker.sh  uuid
balancer_test.go  circle.yml      CONTRIBUTING.md           errors.go  influxvar.go                meta      package.sh  scripts    statik    tests
build-docker.sh   client          Dockerfile                etc        LICENSE                     models    pkg         services   stress    toml
CHANGELOG.md      cluster         Dockerfile_test_ubuntu32  importer   LICENSE_OF_DEPENDENCIES.md  monitor   QUERIES.md  shared     tcp       tsdb
$ cd models/
$ go test -run TestParsePointNoValue
PASS
ok      github.com/influxdb/influxdb/models     0.001s
```
will run the test that is failing on Circle.

**Comment by oiooj** [comment]
No problem here, I don't know what's going on.  Always failed on Circle.
``` shell
Sosara@MBA:~/golang/src/github.com/oiooj/influxdb/models$ go test -run TestParsePointNoValue
PASS
ok      github.com/oiooj/influxdb/models        0.010s
```

**Comment by oiooj** [comment]
@otoolep fixed , Thanks

**Comment by otoolep** [comment]
Thanks @oiooj.

Can you please rebase and squash the commits, with a single commit?

**Comment by otoolep** [comment]
Can you also add a test case for this? We really need test cases for every change.

**Comment by oiooj** [comment]
rebased the commits , I reviewed `handler_test.go` source, This is a post and write data request, It's difficult to add a test case.

File: services/httpd/handler.go [hunk 1]
```diff
@@ -461,7 +461,7 @@ func (h *Handler) serveWriteLine(w http.ResponseWriter, r *http.Request, body []
 			}
 
 			// check that the byte is in the standard ascii code range
-			if body[i] > 32 {
+			if body[i] > 32 || i >= len(body)-1 {
```
Comment (otoolep) [inline services/httpd/handler.go]: Why not just `i>len(body)`?

Comment (otoolep) [inline services/httpd/handler.go]: Scrap that, not the same thing.

Comment (otoolep) [inline services/httpd/handler.go]: OK, I see what you are doing. Basically, it it's not JSON by index (len-1), it's not JSON at all. Is that what you're getting at?

**Comment by otoolep** [comment]
I believe I follow what this patch fixes, but I'd like a second set of eyes.

@corylanou ?

**Comment by corylanou** [comment]
Yes this appears to make sense.  We should add a test though so that we don't regress.  Once a test is added, +1.  Thanks!

**Comment by otoolep** [comment]
Closing in favour https://github.com/influxdb/influxdb/pull/4625

**Pull request closed without merging**